core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -901,7 +901,7 @@ class DAGScheduler(
       // the stage as completed here in case there are no tasks to run
       markStageAsFinished(stage, None)
 
-      val debugString = stage match {
+      def debugString: String = stage match {
```

Comment:
Even if its not heavy , we can easily make it lazy. 
By passing it directly , it will still always evaluate the expression first.